### ISSUE TITLE
OSX Crosstool cannot build armv7s architecture

### ISSUE THREAD

**Issue by rahul-malik** [bazelbuild/bazel#3289]
Please provide the following information. The more we know about your system and use case, the more easily and likely we can help.

### Description of the problem / feature request / question:

When using the osx crosstool and building for armv7s, we receive a null pointer exception. We don't hit this for arm64 or armv7 architectures. 

### If possible, provide a minimal example to reproduce the problem:

Build a library with the OSX crosstool and armv7s architecture specified by `--ios_multi_cpus`

### Environment info
* Operating System:
macOS Sierra

* Bazel version (output of `bazel info release`):
0.5.2

###  Have you found anything relevant by searching the web?

I have not found anything useful on the web.

### Anything else, information or logs or outputs that would be helpful?

```
➜  ios git:(master) ✗ bazel build [INSERT IOS TARGETS HERE] --ios_minimum_os=9.0 --ios_sdk_version=11.0 --compilation_mode opt --ios_multi_cpus armv7s --verbose_failures --apple_bitcode=embedded --xcode_version=9.0 --apple_generate_dsym

............
Unhandled exception thrown during build; message: null
INFO: Elapsed time: 8.531s
java.lang.NullPointerException
	at com.google.devtools.build.lib.skyframe.SkyframeExecutor.getConfigurations(SkyframeExecutor.java:1373)
	at com.google.devtools.build.lib.analysis.BuildView.getDynamicConfigurations(BuildView.java:908)
	at com.google.devtools.build.lib.analysis.BuildView.nodesForTopLevelTargets(BuildView.java:847)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:476)
	at com.google.devtools.build.lib.buildtool.BuildTool.runAnalysisPhase(BuildTool.java:479)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:194)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:349)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:67)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:525)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:314)
	at com.google.devtools.build.lib.runtime.CommandExecutor.exec(CommandExecutor.java:53)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:849)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$2100(GrpcServerImpl.java:104)
	at com.google.devtools.build.lib.server.GrpcServerImpl$4$1.run(GrpcServerImpl.java:915)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
java.lang.NullPointerException
	at com.google.devtools.build.lib.skyframe.SkyframeExecutor.getConfigurations(SkyframeExecutor.java:1373)
	at com.google.devtools.build.lib.analysis.BuildView.getDynamicConfigurations(BuildView.java:908)
	at com.google.devtools.build.lib.analysis.BuildView.nodesForTopLevelTargets(BuildView.java:847)
	at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:476)
	at com.google.devtools.build.lib.buildtool.BuildTool.runAnalysisPhase(BuildTool.java:479)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:194)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:349)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:67)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:525)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:314)
	at com.google.devtools.build.lib.runtime.CommandExecutor.exec(CommandExecutor.java:53)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:849)
	at com.google.devtools.build.lib.server.GrpcServerImpl.access$2100(GrpcServerImpl.java:104)
	at com.google.devtools.build.lib.server.GrpcServerImpl$4$1.run(GrpcServerImpl.java:915)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by c-parsons** [comment]
There are really two issues here:

1. You can't build with armv7s. This is actually somewhat "intended" (or rather, willingly unsupported), as, by name, it hasn't been supported by default since Xcode 6. From what I understand, armv7s is a superset of armv7 -- can you use armv7 here?
2. Bazel threw a NullPointerException with an unhelpful error message, which shouldn't happen under any cirucmstances.

I'll keep this open to track the latter.

**Comment by rahul-malik** [comment]
@c-parsons - Yes we should be able to just build for `arm64` and `armv7`. The `armv7s` architecture I believe was relevant around the release of the iPhone 5/5c. 

This was confusing because when specifying our architectures (using `--ios_multi_cpus`) we've always supplied `armv7s` and didn't encounter an error. When we adopted the crosstool (and upgraded to 0.5.2) we started getting this exception.
